tests/model/model_flattening.cpp

```diff
@@ -591,393 +591,125 @@ TEST(ModelFlattening, importedComponentWithNameClashes)
     EXPECT_EQ(e, a);
 }
 
-// The variable ordering changes when using the HH model defined with imports.
-TEST(ModelFlattening, hodgkinHuxleyDefinedUsingImports)
+TEST(ModelFlattening, importingComponentThatAlsoHasAnImportedComponentAsAChild)
 {
-    auto parser = libcellml::Parser::create();
-    auto model = parser->parseModel(fileContents("modelflattening/hodgkin_huxley_squid_axon_model_1952/model.cellml"));
+    const std::string e =
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+        "<model xmlns=\"http://www.cellml.org/cellml/2.0#\" name=\"a_model\">\n"
+        "  <component name=\"membrane\"/>\n"
+        "  <component name=\"sodium_channel_1\"/>\n"
+        "  <encapsulation>\n"
+        "    <component_ref component=\"membrane\">\n"
+        "      <component_ref component=\"sodium_channel_1\"/>\n"
+        "    </component_ref>\n"
+        "  </encapsulation>\n"
+        "</model>\n";
 
-    EXPECT_TRUE(model->hasUnresolvedImports());
-    model->resolveImports(resourcePath("modelflattening/hodgkin_huxley_squid_axon_model_1952/"));
-    EXPECT_FALSE(model->hasUnresolvedImports());
+    auto model = libcellml::Model::create("a_model");
 
-    model->flatten();
+    // Import the membrane component from a file
+    auto membrane = libcellml::Component::create("membrane");
+    model->addComponent(membrane);
 
-    auto printer = libcellml::Printer::create();
+    auto membraneImporter = libcellml::ImportSource::create();
+    membraneImporter->setUrl("basicmembranemodel.cellml");
 
-    auto a = printer->printModel(model);
-    auto modelNonImportVersion = parser->parseModel(fileContents("generator/hodgkin_huxley_squid_axon_model_1952/model.cellml"));
-    auto e = printer->printModel(modelNonImportVersion);
-    EXPECT_EQ(e, a);
+    membrane->setImportSource(membraneImporter);
+    membrane->setImportReference("membrane_for_importing");
 
-    auto generator = libcellml::Generator::create();
+    // Import the sodium channel from a file, and set as encapsulated child of the membrane
+    auto sodiumChannel = libcellml::Component::create("sodium_channel");
+    membrane->addComponent(sodiumChannel);
 
-    generator->processModel(model);
+    auto sodiumImporter = libcellml::ImportSource::create();
+    sodiumImporter->setUrl("basicsodiumchannel.cellml");
```

Comment:
We are really inconsistent when it comes to our file name convention. Yet, in most cases we use `_` to separate words, so here I would expect to see `basic_sodium_channel.cellml` (something similar for other file names in this PR).

Reply:
Looking at the `modelflattening` folder we're about 50/50 with the naming.  I've changed these new tests by adding the underscore, but wanted to check whether this was a big deal before going through the rest of them, as it affects pretty much every test involving a file and its contents, as well as their names ...?

Reply:
That's the other thing, the `modelflattening` folder should be named `model_flattening` to be consistent with what we do elsewhere. As for the files in that folder being 50/50 with the naming, well... the same person was involved. (Not giving names... :p :))

Anyway, only things related to this PR should be "fixed". Anything else should be done as part of issue #402.

Reply:
It's not just this folder though, it's throughout all the tests.  We have .xml as well as .cellml extensions, camelCase, underscores, etc.  In some ways it's good to have variety of names as it "tests" them all out, but it looks inconsistent.  The reason I stopped renaming all the files here is because a lot of the missing underscores seemed to make sense: eg, the "import_of_reallylongfilename" stuff is easier to read when the "import_of_" looks different from the "reallylongfilename" bit, esp when there are several generations of importing.  So I think that those should stay as they are ...